### PR TITLE
fix: authorization checks on settings-write AJAX handlers + Dokan nonce exposure

### LINKED ISSUE #252: Quick Cart: Plus sign missing
In the quick cart front view of the cart. the plus sign is missing that helps the user to understand to increase the product quantity.
image

### PR DESCRIPTION
**Pull request by kzamanbd**
- Closes getdokan/storegrowth-sales-booster-pro#252
- Closes https://github.com/getdokan/storegrowth-sales-booster-pro/issues/245

## Problem

Every AJAX handler that writes a plugin option was guarded only by `check_ajax_referer( 'spsg_ajax_nonce' )` with **no capability check** before `update_option()`. A nonce proves a request came from a page the site served — not that the sender is allowed to perform the action.

On a Dokan install the frontend vendor dashboard localized the **same** `spsg_ajax_nonce` (`integrations/includes/Dokan/Dashboard/Bogo.php:150`), gated only by `dokan_is_seller_dashboard()` (which page is rendering, not who is viewing). Any user who can load the vendor dashboard obtained a token the privileged settings handlers accept.

## Runtime verification

Tested on dokan-lite 5.0.12, dokan-pro 5.0.11, storegrowth-lite 2.1.1.

**Logged-in non-admin (vendor, role `seller`; `manage_options`=N, `manage_woocommerce`=N) — CONFIRMED privilege escalation.**
POST `admin-ajax.php action=spsg_fly_cart_save_settings` with the dashboard nonce → `HTTP 200 {"success":true}`, global option written. The same nonce validates for every module's `*_save_settings` handler.

**Logged-out visitor at `/dashboard/` — not exploitable.** The storegrowth vendor enqueue does not fire logged-out (no `spsgAdmin`, no nonce), and settings handlers have no `wp_ajax_nopriv_*` registration.

Severity: **authenticated privilege escalation**, not unauthenticated. Rides 2.2 per the issue.

After the fix, the same vendor request returns `HTTP 403 {"success":false,"data":"You are not allowed to perform this action."}`; admin saves are unchanged (`HTTP 200`).

## Changes

- **Capability checks** — `current_user_can( 'manage_options' )` before every option read/write in the settings AJAX handlers: core `includes/Ajax.php` (`admin_ajax`, `spsg_inisetup_flag_update`), all 8 module `Ajax` classes, and the Dokan integration `Ajax`. Matches the 2.1.1 standard already used by sales-pop `create_popup` and bogo `bogo_category_msg_create`. Every StoreGrowth settings page already requires `manage_options` (`AdminMenu`), so no legitimate user loses access.
- **Nonce scope** — stopped localizing `spsg_ajax_nonce` on the frontend vendor dashboard (`Bogo.php`). It was unused there (the dashboard app reads only `isPro` / `buyXGetXEnableForVendor`); removing it closes the leak at source. Vendor operations continue to use the REST (`wp_rest`) and frontend (`spsg_frontend_ajax_nonce`) nonces.
- **Method-dispatch hardening** — the fly-cart nopriv handler replaced its open `method_exists()` dispatch with an explicit allow-list (`get_cart_contents`), filterable via `spsg_fly_cart_frontend_allowed_methods`.

## Enumeration (completeness gate)

Every `wp_ajax_*` / `wp_ajax_nopriv_*` registration was enumerated (handler, file:line, nonce, context, capability check, option written). All 11 unguarded write handlers are now gated; the 4 already-guarded 2.1.1 handlers are unchanged; the nopriv sinks write no options. `quick-view`'s nopriv `ajax_quickview_callback` was reviewed — read-only product render, no method dispatch, no change needed.

## Non-regression

- Admin saves every module's settings through the normal UI — unchanged.
- fly-cart `get_cart_contents` over the nopriv endpoint still works; a disallowed method name is rejected.
- phpcs: zero new violations (diffed against baseline — same pre-existing errors, line numbers shifted only).

> Note: cross-repo `Closes` links the pro issue but GitHub does not auto-close across repositories; close #252 manually on merge.

### CODE REVIEW

**Comment by kzamanbd** [comment]
### Also closes getdokan/storegrowth-sales-booster-pro#245

Issue #245 ("no capability check on any AJAX handler — all modules") is the broad version of this pattern; #252 was its critical subset. This PR now covers the remainder:

- **Capability checks** on every listed settings read/write handler are already in this PR: `spsg_admin_ajax`, `spsg_inisetup_flag_update`, `spsg_bogo_general_save_settings`, and the `save_settings`/`get_settings` pairs for countdown-timer, direct-checkout, floating-notification-bar, fly-cart, progressive-discount-banner, quick-view, stock-bar. `bogo_category_msg_list` already had one (2.1.1); `spsg_process_user_concent_data` was removed in the consent-cleanup PR.
- **Method dispatch:** `spsg_admin_ajax` now uses an explicit allow-list (`get_all_modules`, `update_module_status`) — unknown methods return 400 before `call_user_func`. The fly-cart `nopriv` dispatcher was already allow-listed here.
- **Public storefront handlers** `spsg_fly_cart_frontend` and `spsgqcv_quickview` are documented as intentionally public: frontend-nonce-guarded, read-only / caller's-own-cart, no writes.

**Dokan vendor non-regression audit (the #245 blast-radius concern):** verified no vendor (`dokandar`, not `manage_options`) reaches any newly-gated handler. The vendor dashboard BOGO app uses the REST `VendorBogoController` (dokandar-scoped) via `apiFetch`; the vendor countdown-timer integration saves through `dokan_process_product_meta` (product meta, not the settings ajax); the Dokan fly-cart/countdown settings are `admin_enqueue_scripts` (admin, `manage_options`). So the `manage_options` checks here do not break any vendor workflow.

Runtime-verified: admin `spsg_admin_ajax` with a bogus method → 400; with `get_all_modules` → module list.

**Comment by shohan0120** [comment]
## 🐛 Malformed `form_data` silently wipes two modules' settings and reports success

**Severity:** High (data loss). **Pre-existing**, not introduced by this PR — reported here because
this PR edits both of these `save_settings()` methods (adding the capability guard) at the top of
the very functions that carry the defect.

### Summary
`Floating Notification Bar` and `Progressive Discount Banner` both `json_decode()` a `form_data`
value that they assume is a **JSON string**, then call `update_option()` **unconditionally** with
whatever came out. Any request whose `form_data` is not a JSON string containing `shipping_bar_data`
overwrites the option with an **empty array** — and still returns `{"success":true}`.

### Environment
StoreGrowth Lite 2.1.2 · WP 6.9.x · PHP 8.x · authenticated administrator

### Steps to reproduce
1. Confirm the option has content:
   ```bash
   wp option get spsg_progressive_discount_banner_settings --format=json
   ```
2. As an administrator, POST to `admin-ajax.php` with `form_data` sent as an **array** rather than
   a JSON string (this is what an out-of-date or third-party client sends):
   ```
   action=spsg_progressive_discount_banner_save_settings
   _wpnonce=<valid spsg_ajax_nonce>
   form_data[some_key]=value
   ```
3. Re-read the option.
4. Repeat with `action=spsg_floating_notification_bar_save_settings`.

### Expected
The request is rejected, or at minimum the stored settings are left intact.

### Actual
- Response: `HTTP 200 {"success":true,"data":[]}`
- The option is now `[]` — **all settings destroyed**.
- `wp-content/debug.log` records the cause:
  ```
  PHP Warning:  json_decode() expects parameter 1 to be string, array given in
    .../modules/progressive-discount-banner/includes/Ajax.php on line 44
  PHP Warning:  json_decode() expects parameter 1 to be string, array given in
    .../modules/floating-notification-bar/includes/Ajax.php on line 44
  ```

### Evidence
Both handlers are identical in shape (`Ajax.php:44-53`):
```php
$form_data = isset( $_POST['form_data'] ) ? json_decode( wp_unslash( $_POST['form_data'] ), true ) : array();
$bar_data  = isset( $form_data['shipping_bar_data'] ) ? $form_data['shipping_bar_data'] : array();
// ...
update_option( 'spsg_progressive_discount_banner_settings', $bar_data );   // unconditional
wp_send_json_success( ... );                                              // reports success
```
There is no guard that `$form_data` decoded successfully, and none that `$bar_data` is non-empty.

The normal UI path is **not** affected — saving from the settings screen sends the correct payload
and persists fine (verified: 31 and 32 keys written respectively). The defect only bites on a
malformed payload, which is exactly when data should be protected rather than destroyed.

### Impact
A single bad request from any authenticated admin-capable client silently destroys a module's
entire configuration, with a success response and no way to recover — WordPress options carry no
revision history. I hit this accidentally during testing and lost the real stored value.

### Suggested fix
Bail when the decode fails or the expected key is missing:
```php
if ( ! is_array( $form_data ) || ! isset( $form_data['shipping_bar_data'] ) ) {
    wp_send_json_error( __( 'Invalid settings payload.', 'storegrowth-sales-booster' ), 400 );
}
```

### Screenshot
No meaningful screenshot exists for this one — it is an API-level defect whose evidence is the
HTTP response, the option value before/after, and the `debug.log` warning, all quoted verbatim above.

---
*Filed from the 2.1.2 release-gate run.*
